### PR TITLE
Style edits to meteor-fast

### PR DESCRIPTION
**Pull request by bradcray**
This is my set of proposed cleanups to the faster meteor version:

* removed multi-step history of program because previous program will
  itself embed that, we don't do this for other versions, and in the
  limit, many of our versions would result in a huge link of predecessors.
* replaced a lot of 'var x: int' with 'var x = 0' because the code
  relies on the value to be a '0', the intent seems clearer for
  someone not familiar with Chapel's auto-initialization, and it's
  just as clearly an 'int'.
* changed base-case arguments to searchParallel() into default arguments
* changed some instances of "setup" (a noun) into "set up" (an action)
* changed some vars into consts
* introduced curly brackets for single statement control flow where the
  body of that control flow is itself multi-line control flow.
* replaced unnecessary uses of 'in' intents
* pushed variable/const declarations to first point of definition.
* changed a variable + break into returns
* removed some seemingly unnecessary uint casts
* reformatted a long line
* Flattened an unnecessarily nested conditional

One change I did not make, but was tempted to, was to flatted initial blank lines out of 'proc's which look strange and too space-y to my eye (and that's as someone who's usually a big fan of vertical whitespace.

Added a TODO to the README about serial reductions.

### CODE REVIEW

**Comment by bradcray** [comment]
@ben-albrecht : Well, I got quite far thinking "Gosh, I don't think I really have any proposed changes to make to the code" before the dam broke.  Take a look and see what you think.

File: test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl [hunk 1]
```diff
@@ -358,8 +352,8 @@ proc searchLinearHelper(in board, in pos, in used, in placed,
 //
 record BackoffSpinLock {
   var l: atomic bool,
-      lockAttempts: uint,
-      maxLockAttempts = (2**16-1): uint;
+      lockAttempts = 0,
+      maxLockAttempts = (2**16-1);
```
Comment (ben-albrecht) [inline test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl]: I believe these 2 need to be uint still, for the additional upper bound. @ronawho can confirm this. ( I originally suggested a similar change)

**Comment by ben-albrecht** [comment]
Changes look great other than my uint comment. Were you able to confirm performance remained the same? If so, should be good to merge (after taking my other comment into consideration as well).

File: test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl (hunk 1 above)
Comment (bradcray) [inline test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl]: I'm not understanding what you mean by the additional upper bound.  You're not suggesting we'd make 2**63 attempts to take the lock, are you?

**Comment by bradcray** [comment]
I was going to let the nightly testing check the performance.

File: test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl (hunk 1 above)
Comment (bradcray) [inline test/studies/shootout/meteor/kbrady/meteor-parallel-alt.chpl]: I forgot @ben-albrecht was out today.  Chatting with @ronawho by email, his concern was wraparound.  I convinced him that it wouldn't happen fast enough with a 64-bit int and he realized he'd been thinking of C 32-bit ints.  So I'm going to assume we're good here.

**Comment by bradcray** [comment]
Just compared performance on my linux box and it seems unchanged from the previous study version (which itself is an improvement over the release version).

**Pull request merged**